src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -553,7 +536,21 @@ private unsafe int HandleEventShutdownComplete(ref SHUTDOWN_COMPLETE data)
     {
         if (data.ConnectionShutdown != 0)
         {
-            Exception exception = ThrowHelper.GetConnectionAbortedException(_connectionState.AbortErrorCode);
+            bool shutdownByApp = data.ConnectionShutdownByApp != 0;
+            bool closedRemotely = data.ConnectionClosedRemotely != 0;
+            Exception exception = (shutdownByApp, closedRemotely) switch
+            {
+                // It's local shutdown by app, this side called QuicConnection.CloseAsync, throw QuicError.OperationAborted.
+                (true, false) => ThrowHelper.GetOperationAbortedException(),
+                // It's remote shutdown by app, peer's side called QuicConnection.CloseAsync, throw QuicError.ConnectionAborted.
+                (true, true) => ThrowHelper.GetConnectionAbortedException((long)data.ConnectionErrorCode),
+                // It's local shutdown by transport, assuming idle connection (TODO: we should get Connection.CloseStatus), throw QuicError.ConnectionIdle.
+                // https://github.com/dotnet/runtime/issues/72666
+                (false, false) => ThrowHelper.GetExceptionForMsQuicStatus(QUIC_STATUS_CONNECTION_IDLE),
+                // It's remote shutdown by transport, (TODO: we should propagate transport error code), throw QuicError.InternalError.
+                // https://github.com/dotnet/runtime/issues/72666
+                (false, true) => ThrowHelper.GetExceptionForMsQuicStatus(QUIC_STATUS_INTERNAL_ERROR, $"Shutdown by transport {data.ConnectionErrorCode}"),
```

Comment:
You can totally disregard me complaining, but by the time I got to this line, I forgot completely what these bools were about and in what order they were specified😅 if you are so against the good ol' `if`s, then maybe at least `(shutdownByApp: false, closedRemotely: true)`?

P.S.: first bool going true->false and second going false->true hurts my OCD 😅

Reply:
Ordered and added variable names.

Reply:
Thanks! I don't see the change though 🤔

Reply:
I lost it somewhere...